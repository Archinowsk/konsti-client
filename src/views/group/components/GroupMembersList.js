/* @flow */
import React from 'react'
import { useTranslation } from 'react-i18next'
import type { GroupMember } from 'flow/group.flow'
import type { StatelessFunctionalComponent } from 'react'

type Props = {
  groupMembers: $ReadOnlyArray<GroupMember>,
}

const GroupMembersList: StatelessFunctionalComponent<Props> = (
  props: Props
) => {
  const { groupMembers } = props
  const { t } = useTranslation()

  if (!groupMembers) return <div className='group-members-list' />

  const membersList = groupMembers.map(member => {
    const leader = member.serial === member.playerGroup
    return (
      <React.Fragment key={member.username}>
        <li>
          {member.username} {leader && <span>({t('groupLeader')})</span>}
        </li>
      </React.Fragment>
    )
  })

  return <div className='group-members-list'>{membersList}</div>
}

export default GroupMembersList
