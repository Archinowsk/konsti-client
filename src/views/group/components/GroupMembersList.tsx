import React, { FunctionComponent, ReactElement } from 'react';
import { useTranslation } from 'react-i18next';
import { GroupMember } from 'typings/group.typings';

export interface Props {
  groupMembers: readonly GroupMember[];
}

export const GroupMembersList: FunctionComponent<Props> = (
  props: Props
): ReactElement<'div'> => {
  const { groupMembers } = props;
  const { t } = useTranslation();

  if (!groupMembers) return <div className='group-members-list' />;

  const membersList = groupMembers.map((member, index) => {
    const leader = member.serial === member.groupCode;
    return (
      <p key={member.username}>
        {index + 1}) {member.username}{' '}
        {leader && <span>({t('groupLeader')})</span>}
      </p>
    );
  });

  return <div className='group-members-list'>{membersList}</div>;
};
