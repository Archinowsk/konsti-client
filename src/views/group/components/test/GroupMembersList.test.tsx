import React from 'react';
import { shallow } from 'enzyme';
import { GroupMembersList, Props } from '../GroupMembersList';

const groupMembers = [];

describe('GroupMembersList', () => {
  it('should render correctly', () => {
    const props: Props = { groupMembers };
    const component = shallow(<GroupMembersList {...props} />);
    expect(component).toMatchSnapshot();
  });
});
