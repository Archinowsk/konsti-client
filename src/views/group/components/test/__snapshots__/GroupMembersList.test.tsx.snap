// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupMembersList should render correctly 1`] = `
<div
  className="group-members-list"
/>
`;
